File: CarRentalFrontEnd/src/services/bookingService.js
import api from './api';

// Creates a new booking
export const createBooking = async (bookingData) => {
    try {
        const response = await api.post('/bookings', bookingData);
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};

// Fetches bookings for the currently logged-in user
export const getMyBookings = async () => {
    try {
        const response = await api.get('/bookings/my-bookings');
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};

export const cancelBooking = async (bookingId) => {
    try {
        const response = await api.put(`/bookings/${bookingId}/cancel`);
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};

export const getAllBookings = async () => {
    try {
        const response = await api.get('/bookings');
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};

// Updates the status of a booking (Admin only)
export const updateBookingStatus = async (bookingId, status) => {
    try {
        const response = await api.put(`/bookings/${bookingId}/status/${status}`);
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};

export const getBookingById = async (bookingId) => {
    try {
        const response = await api.get(`/bookings/${bookingId}`);
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};

File: CarRentalBackEnd/src/main/java/com/carRental/service/BookingService.java
package com.carRental.service;

import java.util.List;

import com.carRental.dto.BookingRequestDTO;
import com.carRental.dto.BookingResponseDTO;
import com.carRental.dto.BookingStatusDTO;

// Service interface for managing car bookings
public interface BookingService {

    // Creates a new booking
    BookingResponseDTO createBooking(BookingRequestDTO bookingRequestDTO);

    // Retrieves booking details by ID
    BookingResponseDTO getBookingById(Long id);

    // Retrieves all bookings (Admin only)
    List<BookingResponseDTO> getAllBookings();

    // Retrieves bookings for a specific customer
    List<BookingResponseDTO> getBookingsByCustomer(Long customerId);

    // Retrieves bookings for the currently logged-in user
    List<BookingResponseDTO> getBookingsForLoggedInUser();

    // Gets the current status of a booking
    BookingStatusDTO getBookingStatus(Long bookingId);

    // Updates the status of a booking (e.g., CONFIRMED, CANCELLED)
    BookingResponseDTO updateBookingStatus(Long bookingId, String status);

    // Cancels an existing booking
    BookingResponseDTO cancelBooking(Long bookingId);
}
